.github/workflows/prerelease_macos.yml

```diff
@@ -0,0 +1,119 @@
+name: Prerelease macOS pipeline
+
+on:
+  release:
+    types:
+      - prereleased
+    tags:
+      - '*'
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GO_VERSION: '1.14.4'
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+  GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
+  SNYK_TOKEN: ${{ secrets.CAOS_SNYK_TOKEN }}
+  TAG:  ${{ github.event.release.tag_name }}
+  DOCKER_HUB_ID: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+  DOCKER_HUB_PASSWORD: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+  SCHEMA_BRANCH: ${{ github.event.release.tag_name }}
+  AWS_S3_BUCKET_NAME: "nr-downloads-ohai-staging"
+  AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock-staging"
+  AWS_REGION: "us-east-1"
+  DOCKER_PUBLISH: true
+
+jobs:
+  unit-test-macos:
+    name: macOS unit tests
+    runs-on: macos-11
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{env.GO_VERSION}}
+
+      - name: Running unit tests
+        run: make test-coverage
+
+      - name: Coveralls Parallel
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          path-to-lcov: lcov.info
+          flag-name: run-macos
+          parallel: true
+
+  finish:
```

Comment:
I was thinking, do we really need coveralls for pre/release?

Reply:
Not really. We have them for other OSses prerelease though.

Reply:
Wait, yes we need these, as we don't have Coveralls for release workflow.

Reply:
I thought this is for PRs, no?

Reply:
we also want to keep track of coverage per release and release candidate